### DISCUSSION TITLE
call function from scene

### DISCUSSION THREAD

**Discussion by jimmy008002** [replit/kaboom#446] [Q&A]
Hi, I am a beginner, I want to call the function outside the scene, is it possible?
For example, 

k.scene("game", () => {

 function jump() {
          if (player.isGrounded()) {
              player.jump(JUMP_FORCE);
              console.log("sdasdsdasdasdaaaaaaaaaaaaaaaaaaaaaaaa")
          }
      }
  
      // jump when user press space
      k.onKeyPress("space", jump);
      k.onClick(jump);

    });

I want to use jump() outside the scene to control the character in-game scene. How can I do it?
Thx

**Comment by jimmy008002** [comment]
Actually, I am doing pose detection to control a game character, but putting everything on one web page include cam, game, detection is out of memory, I can fix it by setting the game size to 360 x 240, otherwise it will stop working, but i need to tackle the problem of controlling first.

**Comment by lajbel** [comment]
> Hi, I am a beginner, I want to call the function outside the scene, is it possible?

Hi!! Welcome to kaboom. You can define your function in the top of your file

```js
kaboom();

function jump(player, JUMP_FORCE) {
   if (player.isGrounded()) {
      player.jump(JUMP_FORCE);
      console.log("sdasdsdasdasdaaaaaaaaaaaaaaaaaaaaaaaa");
   }
}

scene("start", () => {
    const player = add([]);
    jump(player);
});

scene("end", () => {
    const player = add([]);
    jump(player);
});
```

**Comment by jimmy008002** [comment]
Hi thank you so much for answering me!

` k.scene("game", () => {
  
      // define gravity
      k.gravity(2400);
  
      // add a game object to screen
      const player = k.add([
          // list of components
          k.sprite("bean"),
          k.pos(80, 40),
          k.area(),
          k.body(),
      ]);
  
      // floor
      k.add([
        k.rect(k.width(), FLOOR_HEIGHT),
        k.outline(4),
        k.pos(0, k.height()),
        k.origin("botleft"),
        k.area(),
        k.solid(),
        k.color(127, 200, 255),
      ]);
  
       var jump=function() {
          if (player.isGrounded()) {
              player.jump(JUMP_FORCE);
             
          }
      }
  
      // jump when user press space
      k.onKeyPress("space", jump);
      k.onClick(jump);
  
      function spawnTree() {
  
          // add tree obj
          k.add([
            k.rect(48, k.rand(32, 96)),
            k.area(),
            k.outline(4),
            k.pos(k.width(), k.height() - FLOOR_HEIGHT),
            k.origin("botleft"),
            k.color(255, 180, 255),
            k.move(k.LEFT, SPEED),
              "tree",
          ]);
  
          // wait a random amount of time to spawn next tree
          k.wait(k.rand(0.5, 1.5), spawnTree);
  
      }
  
      // start spawning trees
      spawnTree();
  
      // lose if player collides with any game obj with tag "tree"
      player.onCollide("tree", () => {
          // go to "lose" scene and pass the score
          k.go("lose", score);
          k.burp();
          k.addKaboom(player.pos);
      });
  
      // keep track of score
      let score = 0;
  
      const scoreLabel = k.add([
        k.text(score),
        k.pos(24, 24),
      ]);
  
      // increment score every frame
      k.onUpdate(() => {
          score++;
          scoreLabel.text = score;
      });
  
  
      
      
    }
    );

    function jump(player,JUMP_FORCE) {
      if (player.isGrounded()) {
          player.jump(JUMP_FORCE);
         
      }
  }
    
  jump();`


now the problem is Cannot read properties of undefined (reading 'isGrounded') , I try to code in react, is it the reason I need to declare k in front of function?

**Reply by lajbel** [reply]
yeah u need set the player parameter in your function I think
